src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,35 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        switch (type.getSimpleName()) {
+            case "Truck":
+                return new ArrayList<>(new TruckProduce().get());
+            case "Excavator":
+                return new ArrayList<>(new ExcavatorProduce().get());
+            case "Bulldozer":
+                return new ArrayList<>(new BulldozerProducer().get());
+            default:
+                return new ArrayList<>();
+        }
```

Comment:
I'd like to leave here several points: 
- you are creating new producers each time the method is called. 
- I'd recommend creating a map as a class field (using class as a key and producer as a value) and using this map instead of switch
- let's use Collections.emptyList() in case we are going to return empty list